commit 2ca42af6c07afed6a5e94e8e3c889c728869fd2f
Author: Samuel Thibault <samuel.thibault@ens-lyon.org>
Date:   Mon Oct 17 02:30:14 2016 +0200

    Fix too weak references when linking libpthread statically
    
    * sysdeps/pthread/libc-lockP.h [IS_IN (libpthread)]: Do not make __pthread
    symbols weak.

diff --git a/libpthread/sysdeps/pthread/libc-lockP.h b/libpthread/sysdeps/pthread/libc-lockP.h
index 2631e0c..c04881d 100644
--- a/libpthread/sysdeps/pthread/libc-lockP.h
+++ b/libpthread/sysdeps/pthread/libc-lockP.h
@@ -106,7 +106,7 @@ extern int __pthread_atfork (void (*__prepare) (void),
 
 /* Make the pthread functions weak so that we can elide them from
    single-threaded processes.  */
-#ifndef __NO_WEAK_PTHREAD_ALIASES
+#if !defined(__NO_WEAK_PTHREAD_ALIASES) && !IS_IN (libpthread)
 # ifdef weak_extern
 weak_extern (__pthread_mutex_init)
 weak_extern (__pthread_mutex_destroy)
